lib/internal/modules/cjs/loader.js

```diff
@@ -828,11 +828,27 @@ Module.runMain = function() {
   process._tickCallback();
 };
 
-Module.createRequireFromPath = (filename) => {
+function createRequireFromPath(filename) {
   const m = new Module(filename);
   m.filename = filename;
   m.paths = Module._nodeModulePaths(path.dirname(filename));
   return makeRequireFunction(m);
+}
+
+Module.createRequireFromPath = createRequireFromPath;
+
+Module.createRequire = function createRequire(filename) {
+  let path;
+  try {
+    path = fileURLToPath(filename);
+  } catch (e) {
+    if (e.code === 'ERR_INVALID_URL_SCHEME') {
```

Comment:
`fileURLToPath` could throw a `ERR_INVALID_FILE_URL_PATH` error here, which seems like it would then cause an object type error down the line given the fallthrough, which seems non-ideal.

I do think we might need to be a bit stricter here.

How about a logic like:

If `filename` is a URL object or file: string (`typeof filename === 'object' && filename instanceof URL || typeof filename === 'string' && filename.startsWith('file:')`), then pass it to `fileURLToPath` before the `createRequireFromPath` function. Any error can then be fully propagated without a try-catch guard.

Reply:
This logic didn't exactly work because it is possible to have url type strings e.g. `'html:` that will pass through and then not fail early enough.

I've updated this PR to require absolute paths... so the logic here was simplified a bit